addons/l10n_dk/models/account_move.py

```diff
@@ -0,0 +1,21 @@
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_dk_currency_rate_at_transaction = fields.Float(
+        string='Rate',
+        compute='_compute_currency_rate_at_transaction', readonly=True, store=True,
+        digits=(12, 6),
+    )
+
+    @api.depends('currency_id', 'invoice_date', 'date', 'company_id')
```

Comment:
add depend amount_currency

Reply:
I'm not sure about that. We don't use it at all in the computation. What about changing the compute function to compute by ourselves the rate by using `amount_total_signed` and `amount_total`?